### PR TITLE
#163426634 Email and InApp Notification

### PR DESCRIPTION
**Pull request by eokwukwe**
#### What does this PR do?
Email and InApp Notification
#### Description of Task to be completed?
- [x] user can opt in or out for email notification
- [x] user receives notification for new articles
- [x] user receives notification for a comment on articles
- [x] user receives notification for a like on articles
#### How should this be manually tested?
- [x] clone the repo
- [x] cd into the folder and run `npm install` or `yarn install` to install dependencies
- [x] run `npm run test` or `yarn test`
#### What are the relevant pivotal tracker stories?
[#163426634](https://www.pivotaltracker.com/story/show/163426634)

#### Screenshot
screen shot 2019-03-07 at 9 18 44 am
screen shot 2019-03-07 at 9 19 21 am

### CODE REVIEW

File: server/controllers/article.js [hunk 1]
```diff
@@ -2,9 +2,11 @@ import { Op } from "sequelize";
 import isEmpty from "lodash.isempty";
 
 import models from "@models";
+import { getBaseUrl } from "@helpers/users";
 import { httpResponse, serverError } from "@helpers/http";
 import { pagination } from "@helpers/pagination";
 import { generateUniqueSlug, calculateReadTime } from "@helpers/articles";
+import { sendPublishedArticleNotification } from "./notification";
```
Comment (obasajujoshua31) [inline server/controllers/article.js:9]: can you use our module resolver please

File: server/controllers/comment.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import models from "@models";
 import { httpResponse, serverError } from "@helpers/http";
+import { sendCommentNotification } from "./notification";
```
Comment (obasajujoshua31) [inline server/controllers/comment.js:3]: please can you use our module resolve.

File: server/controllers/comment.js [hunk 3]
```diff
@@ -11,16 +12,17 @@ const { Comments } = models;
...
+      articleId: req.article.id,
+      userId: req.user.userId,
+      content: req.body.content
     });
-
+    await sendCommentNotification(
+      req.article.authorId,
+      req.article.id,
+      req.article.title
+    );
```
Comment (obasajujoshua31) [inline server/controllers/comment.js]: please can you consider destructuring.

File: server/controllers/article.js (hunk 1 above)
Comment (eokwukwe) [reply server/controllers/article.js:9]: Module resolver does not affect files that are in the same folder. You can take a look in the `index.js` file in the middlewares folder.

File: server/controllers/comment.js (hunk 2 above)
Comment (eokwukwe) [reply server/controllers/comment.js:3]: Module resolver does not affect files that are in the same folder. You can take a look in the `index.js` file in the middlewares folder.

**Review by ghost — APPROVED** [review]

File: server/helpers/constants.js [hunk 4]
```diff
@@ -2,3 +2,11 @@ export const PAGE_LIMIT_DEFAULT = 20;
 export const PAGE_NUMBER_DEFAULT = 1;
 export const ARTICLE_REACTION = 1;
 export const COMMENT_REACTION = 2;
+export const NOTIFICATION = {
+  publishArticle: "publishArticle",
+  comment: "commentOnArticle",
+  folllowUser: "follower",
```
Comment (obasajujoshua31) [inline server/helpers/constants.js:8]: please your spelling of folllowUser

**Review by obasajujoshua31 — APPROVED** [review]

**Pull request merged**